torch/distributed/distributed_c10d.py

```diff
@@ -496,7 +499,11 @@ def init_process_group(backend,
     # barrier at the end to ensure that once we return from this method, all
     # process groups including global variables are updated correctly on all
     # ranks.
-    if backend == Backend.MPI:
+    if backend == Backend.MPI or not (
+        isinstance(store, TCPStore) or
+        isinstance(store, FileStore) or
+        isinstance(store, HashStore)
```

Comment:
nit: add these to a list `store_allowlist` and then just check `type(store) not in store_allowlist` so we don't duplicate the isinstance checks below?